src/node_mime.cc

```diff
@@ -0,0 +1,369 @@
+      left = right;
+    }
+    bool saw_invalid_value = false;
+    std::string parameterValue;
+    if (right < src.size()) {
+      if (src[right] == '"') {
+        right++;
+        while (true) {
+          left = right;
+          for (; right < src.size(); right++) {
+            auto c = src[right];
+            if (c == '"' || c == '\\') {
+              break;
+            } else if (!is_http_quoted_string_token(c)) {
+              saw_invalid_value = true;
+            }
+          }
+          parameterValue += src.substr(left, right - left);
+          if (right < src.size() && src[right] == '\\') {
+            right++;
+            if (right < src.size()) {
+              parameterValue += src[right];
+              right++;
+              continue;
+            } else {
+              parameterValue += '\\';
+              break;
+            }
+          } else {
+            break;
+          }
+        }
+        for (; right < src.size(); right++) {
+          auto c = src[right];
+          if (c == ';') {
+            break;
+          }
+        }
+      } else {
+        for (; right < src.size(); right++) {
+          auto c = src[right];
+          if (c == ';') {
+            break;
+          } else {
+            auto is_2_byte = (c & (unsigned char)0xE0) == 0xC0;
+            auto is_3_byte = (c & (unsigned char)0xF0) == 0xE0;
+            auto is_4_byte = (c & (unsigned char)0xF8) == 0xF0;
+            bool is_invalid = false;
+            if (is_2_byte) {
+              // need 2 bits of FF
+              if ((c & (unsigned char)0x1C) != 0) {
+                is_invalid = true;
+                right += 1;
+              } else {
+                right += 1;
+                if (right < src.size()) {
+                  c = src[right];
+                  if ((c & 0xC0) != 0x80) {
+                    is_invalid = true;
+                  }
+                } else {
+                  is_invalid = true;
+                }
+              }
+            } else if (is_3_byte) {
+              is_invalid = true;
+              right += 2;
+            } else if (is_4_byte) {
+              is_invalid = true;
+              right += 3;
+            } else if (!is_http_quoted_string_token(c)) {
+              is_invalid = true;
+            }
+            if (is_invalid && !saw_invalid_char) {
+              saw_invalid_value = true;
+              first_invalid_char = right;
+            }
+          }
+        }
+        {
+          auto trim_right = right;
+          while (is_ascii_whitespace(src[trim_right - 1])) {
+            trim_right--;
+          }
+          if (!saw_invalid_char || first_invalid_char >= trim_right) {
+            parameterValue = src.substr(left, trim_right - left);
+          }
+        }
+      }
+    }
+    if (parameterName.size() > 0 &&
+        parameterValue.size() > 0 &&
+        !saw_invalid_value) {
+      parameters.push_back(std::pair<std::string, std::string>(
```

Comment:
Using `std::make_pair(parameter_name, parameter_value)` might also work and be a bit shorter, or just `parameters.push_back({ parameter_name, parameter_value })`

Reply:
fixed